src/OpenTelemetry/Logs/Options/OpenTelemetryLoggerOptions.cs

```diff
@@ -29,29 +30,57 @@ public class OpenTelemetryLoggerOptions
         internal readonly List<BaseProcessor<LogRecord>> Processors = new();
         internal ResourceBuilder ResourceBuilder = ResourceBuilder.CreateDefault();
 
+        /// <summary>
+        /// Gets or sets a value indicating whether or not formatted log message
+        /// should be included on generated <see cref="LogRecord"/>s. Default
+        /// value: <see langword="false"/>.
+        /// </summary>
+        /// <remarks>
+        /// Note: When set to <see langword="false"/> a formatted log message
+        /// will not be included if a message template can be found. If a
+        /// message template is not found, a formatted log message is always
+        /// included.
+        /// </remarks>
+        public bool IncludeFormattedMessage { get; set; }
+
         /// <summary>
         /// Gets or sets a value indicating whether or not log scopes should be
         /// included on generated <see cref="LogRecord"/>s. Default value:
-        /// False.
+        /// <see langword="false"/>.
         /// </summary>
         public bool IncludeScopes { get; set; }
 
         /// <summary>
-        /// Gets or sets a value indicating whether or not formatted log message
-        /// should be included on generated <see cref="LogRecord"/>s. Default
-        /// value: False.
+        /// Gets or sets a value indicating whether or not log state should
+        /// be included on generated <see cref="LogRecord"/>s. Default value:
+        /// <see langword="true"/>.
         /// </summary>
-        public bool IncludeFormattedMessage { get; set; }
+        public bool IncludeState { get; set; } = true;
```

Comment:
Does it make sense to let feedback drive the need for this option? I like the new behavior or ParseStateValues - it's name is still unfortunate being ILoggery, but I'd prefer to avoid adding another.

Reply:
I'm not passionate that we absolutely need this. The idea here is that on this PR I'm redefining (essentially) what `ParseStateValues` means. If it was `false` previously we wouldn't look at the state at all and we wouldn't buffer anything for batch cases. With the redefinition, we automatically process state and also automatically buffer things. So I introduced `IncludeState` to enable what is possible today which is disable the processing & buffering.

Reply:
It looks like we would only set the `state` in the newly added `ProcessState` method with the requirement that `ParseStateValue` is also set to `false` otherwise we could run into situations where `state` is `null` even with `IncludeState` set to `true` which feels a bit weird.

If we want the new default behavior to always generate attributes, then we could achieve that while continuing to respect the older options. Essentially, we would still only assign one of these: `State` or `StateValues` depending on `ParseStateValues` option but we always generate attributes.

Reply:
> We could run into situations where state is null even with IncludeState set to true which feels a bit weird.

This is totally by design. `State` is obsolete. No one should rely on it or use it going forward. It is only set at all to preserve the existing behavior(s) we have today. Maybe it would help to rename `IncludeState` -> `IncludeAttributes`? I went ahead and made this change so we could see how it might look. Also updated XML comments to try and document all the behaviors.

> Essentially, we would still only assign one of these: State or StateValues depending on ParseStateValues option but we always generate attributes.

I suppose we could do that. But consider an exporter today like `GenevaExporter`. It checks `State` and then `StateValues`. Let's say `ParseStateValues = false`. So we set `State = state, StateValues = null` as we do today and then we automatically set `Attributes` to something useful (new behavior). `GenevaExporter` now needs an update to take advantage of this. The current behavior on the PR is `State = null, StateValues = Attributes = usefulStuff` so it will just fall into its existing `StateValues` branch and doesn't really need to be touched to get the new useful things.

Reply:
I think it's better to make the exporters update their implementation to start using new features. I feel by marking `State` and `StateValues` oboslete, we essentially want to discourage exporters from using the obsolete properties and use the newer spec complaint properties instead such as `Body` and `Attributes`.

Why make it easy for them to continue consuming obsolete properties by providing them newer features through the obsolete properties?

Reply:
Interesting perspective. Thinking I would prefer to just drop `Attributes` and continue along with `StateValues`. `Attributes` really isn't a new feature, it is just renaming `StateValues` to be more compliant with the spec. I like your idea to encourage exporters to update but I don't think this is the best way to go about it. Introducing differences between `StateValues` and `Attributes` just seems like unnecessary confusion and documentation burden 😄

`Body` is a new feature for sure. And it more fits into what you are saying. Exporters have to update if they want to take advantage of it.

Reply:
Let's hold off on adding a new configuration option. There's a handful of things going on here.

1) New behavior to StateValues
2) Obsoleting StateValues and introducing Attributes to be the replacement it
3) Introducing IncludeAttributes for configuring old behavior of not processing them by default

I see each of these as a separate PR. I support the first item being in this PR. I also favor doing the second item, but a separate PR would nicely document our conversation and ultimate decision. I'm not in favor of the third item, but happy to continue a focused conversation on a separate PR that introduces it.